Pat.Sender/TopicClientResolver.cs

```diff
@@ -28,5 +30,25 @@ public static TopicClient GetTopic(string connectionString, string topicName)
                 return topicClient;
             }
         }
+
+        public static TopicClient GetTopic(string connectionString, string topicName, ITokenProvider tokenProvider)
+        {
+            if (TokenClients.ContainsKey(connectionString))
+            {
+                return TokenClients[connectionString];
+            }
+
+            lock (Locks.GetOrAdd(connectionString, new object()))
+            {
+                if (TokenClients.ContainsKey(connectionString))
+                {
+                    return TokenClients[connectionString];
+                }
+                ServiceBusConnectionStringBuilder builder = new ServiceBusConnectionStringBuilder(connectionString);
+                var topicClient = new TopicClient(builder.Endpoint, topicName, tokenProvider,retryPolicy: RetryPolicy.Default);
```

Comment:
Formatting, if you press ctrl + K, ctrl + D in VS it'll format the whole file for you